Tomcat startup fails if citation-page.enabled_communities is set

## Describe the bug

if _citation-page.enabled_communities_ configuration option is set (=nonempty value), Tomcat (or standalone server-boot.jar) fails to start, resulting to stack trace as follows:

`
ERROR unknown unknown org.springframework.boot.SpringApplication @ Application run failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'bitstreamRestController': Unsatisfied dependency expressed through field 'citationDocumentService': Error creating bean with name 'org.dspace.disseminate.CitationDocumentServiceImpl#0' defined in file [...\dspace\config\spring\api\core-services.xml]: null
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:788)
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:768)
        at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:146)
        at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:509)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1459)

Caused by: java.lang.UnsupportedOperationException
        at java.base/java.util.AbstractList.add(AbstractList.java:153)
        at java.base/java.util.AbstractList.add(AbstractList.java:111)
        at org.dspace.disseminate.CitationDocumentServiceImpl.afterPropertiesSet(CitationDocumentServiceImpl.java:147)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1873)
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1822)
`

Startup fails regardless the community handle used is valid or not, and regardless of the presence of another configuration option _citation-page.enabled_collections_

The problem is present at least in DSpace 7 and newer versions (tested with 10.0 snapshot as well), so it is not dependent on individual CoverPage implementation (changed in DSpace 9) but due to the fact that an immutable list is currently used as a base for collection list, and thus, it cannot be modified when collections for the configured communit(ies) are added to the list on startup..

## To Reproduce
Steps to reproduce the behavior:

Just add citation-page.enabled_communities with some existing community handle to local.cfg and try to start Tomcat.

## Expected behavior

Tomcat should start and CoverPage should be present in collection under the configured community.

--- a/dspace-api/src/main/java/org/dspace/disseminate/CitationDocumentServiceImpl.java
+++ b/dspace-api/src/main/java/org/dspace/disseminate/CitationDocumentServiceImpl.java
@@ -126,7 +126,7 @@ public class CitationDocumentServiceImpl implements CitationDocumentService, Ini
         //Load enabled collections
         String[] citationEnabledCollections = configurationService
                 .getArrayProperty("citation-page.enabled_collections");
-        citationEnabledCollectionsList = Arrays.asList(citationEnabledCollections);
+        citationEnabledCollectionsList = new ArrayList<String>(Arrays.asList(citationEnabledCollections));
 
         //Load enabled communities, and add to collection-list
         String[] citationEnabledCommunities = configurationService

